docs/source/cpp/streaming_execution.rst

```diff
@@ -175,9 +175,607 @@ their completion::
+    
+    // create a source node
+    ARROW_ASSIGN_OR_RAISE(arrow::compute::ExecNode * source,
+                            arrow::compute::MakeExecNode("source", plan.get(), {}, 
+                            source_node_options));
+
+``FilterNode``
+--------------
+
+:class:`FilterNode`, as the name suggests, provide a container to define a data filtering criteria. 
+Filter can be written using :class:`arrow::compute::Expression`. For instance if the row values
+of a particular column needs to be filtered by a boundary value, ex: all values of column b
+greater than 3, can be written using :class:`arrow::compute::FilterNodeOptions` as follows::
+
+    // a > 3
+    arrow::compute::Expression filter_opt = arrow::compute::greater(
+                                  arrow::compute::field_ref("a"), 
+																	arrow::compute::literal(3));
+
+Using this option, the filter node can be constructed as follows::																
+
+    // creating filter node
+    arrow::compute::ExecNode* filter;
+      ARROW_ASSIGN_OR_RAISE(filter, arrow::compute::MakeExecNode("filter", 
+                            // plan
+                            plan.get(),
+                            // previous input
+                            {scan}, 
+                            //filter node options
+                            arrow::compute::FilterNodeOptions{filter_opt}));
+
+``ProjectNode``
+---------------
+
+:class:`ProjectNode` executes expressions on input batches and produces new batches. 
+Each expression will be evaluated against each batch which is pushed to this 
+node to produce a corresponding output column. This is exposed via 
+:class:`arrow::compute::ProjectNodeOptions` component which requires, 
+a :class:`arrow::compute::Expression`, names of the project columns (names are not provided, 
+the string representations of exprs will be used) and a boolean flag to determine 
+synchronous/asynchronous nature (by default asynchronous option is set to `true`). 
+
+Sample Expression for projection::
+
+    // a * 2 (multiply values in a column by 2)
+    arrow::compute::Expression a_times_2 = arrow::compute::call("multiply", 
+							{arrow::compute::field_ref("a"), arrow::compute::literal(2)});
+
+
+Creating a project node::
+
+    arrow::compute::ExecNode *project;
+        ARROW_ASSIGN_OR_RAISE(project, 
+            arrow::compute::MakeExecNode("project", 
+            // plan
+            plan.get(),
+            // previous node 
+            {scan},
+            // project node options 
+            arrow::compute::ProjectNodeOptions{{a_times_2}}));
+
+``ScalarAggregateNode``
+-----------------------
+
+:class:`ScalarAggregateNode` is an :class:`ExecNode` which provides various 
+aggregation options. The :class:`arrow::compute::AggregateNodeOptions` provides the 
+container to define the aggregation criterion. These options can be 
+selected from `arrow::compute` options. 
+
+1. `ScalarAggregateOptions`
```

Comment:
We shouldn't talk about individual options classes here. We should link to the relevant documentation instead. Most of this should actually be in the API docs, and if not, should be moved to documentation comments in the source code.